crypto/random/rand_utils.go

```diff
@@ -49,3 +53,27 @@ func EvaluateDistributionUniformity(t *testing.T, distribution []float64) {
 	mean := stat.Mean(distribution, nil)
 	assert.Greater(t, tolerance*mean, stdev, fmt.Sprintf("basic randomness test failed: n: %d, stdev: %v, mean: %v", len(distribution), stdev, mean))
 }
+
+// computes a bijection from the set of all permutations
+// into the the set [0, n!-1] (where `n` is the size of input `perm`).
+// input `perm` is assumed to be a correct permutation of the set [0,n-1]
+// (not checked in this function).
+func EncodePermutation(perm []int) int {
+	r := make([]int, len(perm))
```

Comment:
```suggestion
	// We first construct r, which contains elements with the following constraints:
	//   r = { r(0) ∈ [0,n-1], r(1) ∈ [0,n-2], ... , r(n-1) ∈ [0,1], r(n) ∈ [0] }
	r := make([]int, len(perm))
```

Reply:
The suggested comment is the start of explaining what a Lehmer code is. However it doesn't fully explain how `r` is obtained, or why there is a bijection between all permutations and the set `[0,n-1] x [0,n-2] x ... x [0,1] x [0]`, so IMO it doesn't add much information. 

In the original comment I only mentioned "compute Lehmer code", maybe I can add a link to what Lehmer code is (that contains all the details).

Reply:
https://github.com/onflow/flow-go/pull/4844/commits/03b9532779fe2e8c9504bc3472e9131dfc7d0c1f